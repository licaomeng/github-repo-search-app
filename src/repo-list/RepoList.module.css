.repoList {
    max-width: 1000px;
    position: relative;
    padding-inline-start: 20px;
    padding-inline-end: 20px;
    display: block;
    margin-left: auto;
    margin-right: auto;
}

.repoListItem {
    border-top: 1px solid var(--color-border-default);
    display: flex;
    flex-direction: column;
    padding: 15px 0;
}

.infoBar {
    display: flex;
    gap: 15px;
    font-size: 0.8em;
    margin-inline-start: 18px;
}

.repoLanguageColor {
    position: relative;
    top: 1px;
    display: inline-block;
    width: 12px;
    height: 12px;
    border-radius: 50%;
}

.repoList li p {
    display: flex;
    align-items: center;
    margin: 6px 0;
    gap: 3px;
}

.repoList li a {
    color: var(--color-accent-fg);
    text-decoration: none;
}

.repoList li .desc {
    margin-inline-start: 18px;
    overflow: hidden;
    text-overflow: ellipsis;
    display: -webkit-box;
    -webkit-line-clamp: 2;
    -webkit-box-orient: vertical;
}