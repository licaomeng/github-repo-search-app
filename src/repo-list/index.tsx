import * as React from "react";
import styles from "./RepoList.module.css";

type dataList = {
    name?: string
    full_name: string
    html_url: string
    description: string
    updated_at: string
    stargazers_count: number
    language: string
}[];

interface ListProps {
    data: dataList
}

function stringToColor(str: string) {
    let hash = 0;
    let color = "#";

    str = String(str);

    for (let i = 0; i < str.length; i++) {
        hash = str.charCodeAt(i) + ((hash << 5) - hash);
    }

    for (let i = 0; i < 3; i++) {
        const value = (hash >> (i * 8)) & 0xFF;

        color += ("00" + value.toString(16)).substr(-2);
    }

    return color;
}

export function RepoList({ data }: ListProps) {
    return <ul className={styles.repoList}>{
        data.map(item =>
            <li className={styles.repoListItem}>
                <p>
                    <svg aria-hidden="true" height="16" viewBox="0 0 16 16" version="1.1" width="16" data-view-component="true">
                        <path fill-rule="evenodd" d="M2 2.5A2.5 2.5 0 014.5 0h8.75a.75.75 0 01.75.75v12.5a.75.75 0 01-.75.75h-2.5a.75.75 0 110-1.5h1.75v-2h-8a1 1 0 00-.714 1.7.75.75 0 01-1.072 1.05A2.495 2.495 0 012 11.5v-9zm10.5-1V9h-8c-.356 0-.694.074-1 .208V2.5a1 1 0 011-1h8zM5 12.25v3.25a.25.25 0 00.4.2l1.45-1.087a.25.25 0 01.3 0L8.6 15.7a.25.25 0 00.4-.2v-3.25a.25.25 0 00-.25-.25h-3.5a.25.25 0 00-.25.25z"></path>
                    </svg>
                    <a href={item.html_url}>{item.full_name}</a>
                </p>
                <p className={styles.desc}>{item.description}</p>
                <div className={styles.infoBar}>
                    <p>
                        <svg aria-label="star" role="img" height="16" viewBox="0 0 16 16" version="1.1" width="16" data-view-component="true">
                            <path fill-rule="evenodd" d="M8 .25a.75.75 0 01.673.418l1.882 3.815 4.21.612a.75.75 0 01.416 1.279l-3.046 2.97.719 4.192a.75.75 0 01-1.088.791L8 12.347l-3.766 1.98a.75.75 0 01-1.088-.79l.72-4.194L.818 6.374a.75.75 0 01.416-1.28l4.21-.611L7.327.668A.75.75 0 018 .25zm0 2.445L6.615 5.5a.75.75 0 01-.564.41l-3.097.45 2.24 2.184a.75.75 0 01.216.664l-.528 3.084 2.769-1.456a.75.75 0 01.698 0l2.77 1.456-.53-3.084a.75.75 0 01.216-.664l2.24-2.183-3.096-.45a.75.75 0 01-.564-.41L8 2.694v.001z"></path>
                        </svg>
                        {item.stargazers_count}
                    </p>
                    <p>
                        <span className={styles.repoLanguageColor} style={{ backgroundColor: stringToColor(item.language) }}></span>
                        {item.language}
                    </p>
                    <p>Updated on {new Date(item.updated_at)?.toDateString()}</p>
                </div>
            </li>
        )
    }</ul>;
}