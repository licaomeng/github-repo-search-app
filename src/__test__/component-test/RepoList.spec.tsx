import * as React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { RepoList } from "../../components/repo-list";

const mockData_1 = [{
    full_name: "marcosesperon/Messi",
    html_url: "https://github.com/marcosesperon/Messi",
    description: "A simple message plugin for jQuery",
    updated_at: "2022-04-26T16:40:47Z",
    stargazers_count: 143,
    language: "CSS"
}];

const mockData_3 = [{
    full_name: "marcosesperon/Messi",
    html_url: "https://github.com/marcosesperon/Messi",
    description: "A simple message plugin for jQuery",
    updated_at: "2022-04-26T16:40:47Z",
    stargazers_count: 143,
    language: "CSS"
}, {
    full_name: "adityadroid/Messio",
    html_url: "https://github.com/adityadroid/Messio",
    description: "An open source messenger app built using flutter",
    updated_at: "2022-11-26T16:40:47Z",
    stargazers_count: 607,
    language: "Dart"
}, {
    full_name: "Soumyajit2825/MESSI_MANIA",
    html_url: "https://github.com/Soumyajit2825/MESSI_MANIA",
    description: "A simple fan made web-page of LIONEL MESSI",
    updated_at: "2022-09-26T16:40:47Z",
    stargazers_count: 8,
    language: "HTML"
}];

test("renders the repo-list component: no data", () => {
    const view = render(<RepoList data={[]}></RepoList>);

    expect(view).toMatchSnapshot();
});

test("renders the repo-list component: 1 item", () => {
    const view = render(<RepoList data={mockData_1}></RepoList>);

    expect(view).toMatchSnapshot();
});

test("renders the repo-list component: 3 item", () => {
    const view = render(<RepoList data={mockData_3}></RepoList>);

    expect(view).toMatchSnapshot();
});