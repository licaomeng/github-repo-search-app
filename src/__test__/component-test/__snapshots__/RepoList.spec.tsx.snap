// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the repo-list component: 1 item 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul>
        <li>
          <p>
            <svg
              aria-hidden="true"
              data-view-component="true"
              height="16"
              version="1.1"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M2 2.5A2.5 2.5 0 014.5 0h8.75a.75.75 0 01.75.75v12.5a.75.75 0 01-.75.75h-2.5a.75.75 0 110-1.5h1.75v-2h-8a1 1 0 00-.714 1.7.75.75 0 01-1.072 1.05A2.495 2.495 0 012 11.5v-9zm10.5-1V9h-8c-.356 0-.694.074-1 .208V2.5a1 1 0 011-1h8zM5 12.25v3.25a.25.25 0 00.4.2l1.45-1.087a.25.25 0 01.3 0L8.6 15.7a.25.25 0 00.4-.2v-3.25a.25.25 0 00-.25-.25h-3.5a.25.25 0 00-.25.25z"
                fill-rule="evenodd"
              />
            </svg>
            <a
              href="https://github.com/marcosesperon/Messi"
            >
              marcosesperon/Messi
            </a>
          </p>
          <p>
            A simple message plugin for jQuery
          </p>
          <div>
            <p>
              <svg
                aria-label="star"
                data-view-component="true"
                height="16"
                role="img"
                version="1.1"
                viewBox="0 0 16 16"
                width="16"
              >
                <path
                  d="M8 .25a.75.75 0 01.673.418l1.882 3.815 4.21.612a.75.75 0 01.416 1.279l-3.046 2.97.719 4.192a.75.75 0 01-1.088.791L8 12.347l-3.766 1.98a.75.75 0 01-1.088-.79l.72-4.194L.818 6.374a.75.75 0 01.416-1.28l4.21-.611L7.327.668A.75.75 0 018 .25zm0 2.445L6.615 5.5a.75.75 0 01-.564.41l-3.097.45 2.24 2.184a.75.75 0 01.216.664l-.528 3.084 2.769-1.456a.75.75 0 01.698 0l2.77 1.456-.53-3.084a.75.75 0 01.216-.664l2.24-2.183-3.096-.45a.75.75 0 01-.564-.41L8 2.694v.001z"
                  fill-rule="evenodd"
                />
              </svg>
              143
            </p>
            <p>
              <span
                style="background-color: rgb(227, 5, 1);"
              />
              CSS
            </p>
            <p>
              Updated on 
              Wed Apr 27 2022
            </p>
          </div>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul>
      <li>
        <p>
          <svg
            aria-hidden="true"
            data-view-component="true"
            height="16"
            version="1.1"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M2 2.5A2.5 2.5 0 014.5 0h8.75a.75.75 0 01.75.75v12.5a.75.75 0 01-.75.75h-2.5a.75.75 0 110-1.5h1.75v-2h-8a1 1 0 00-.714 1.7.75.75 0 01-1.072 1.05A2.495 2.495 0 012 11.5v-9zm10.5-1V9h-8c-.356 0-.694.074-1 .208V2.5a1 1 0 011-1h8zM5 12.25v3.25a.25.25 0 00.4.2l1.45-1.087a.25.25 0 01.3 0L8.6 15.7a.25.25 0 00.4-.2v-3.25a.25.25 0 00-.25-.25h-3.5a.25.25 0 00-.25.25z"
              fill-rule="evenodd"
            />
          </svg>
          <a
            href="https://github.com/marcosesperon/Messi"
          >
            marcosesperon/Messi
          </a>
        </p>
        <p>
          A simple message plugin for jQuery
        </p>
        <div>
          <p>
            <svg
              aria-label="star"
              data-view-component="true"
              height="16"
              role="img"
              version="1.1"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M8 .25a.75.75 0 01.673.418l1.882 3.815 4.21.612a.75.75 0 01.416 1.279l-3.046 2.97.719 4.192a.75.75 0 01-1.088.791L8 12.347l-3.766 1.98a.75.75 0 01-1.088-.79l.72-4.194L.818 6.374a.75.75 0 01.416-1.28l4.21-.611L7.327.668A.75.75 0 018 .25zm0 2.445L6.615 5.5a.75.75 0 01-.564.41l-3.097.45 2.24 2.184a.75.75 0 01.216.664l-.528 3.084 2.769-1.456a.75.75 0 01.698 0l2.77 1.456-.53-3.084a.75.75 0 01.216-.664l2.24-2.183-3.096-.45a.75.75 0 01-.564-.41L8 2.694v.001z"
                fill-rule="evenodd"
              />
            </svg>
            143
          </p>
          <p>
            <span
              style="background-color: rgb(227, 5, 1);"
            />
            CSS
          </p>
          <p>
            Updated on 
            Wed Apr 27 2022
          </p>
        </div>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders the repo-list component: 3 item 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul>
        <li>
          <p>
            <svg
              aria-hidden="true"
              data-view-component="true"
              height="16"
              version="1.1"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M2 2.5A2.5 2.5 0 014.5 0h8.75a.75.75 0 01.75.75v12.5a.75.75 0 01-.75.75h-2.5a.75.75 0 110-1.5h1.75v-2h-8a1 1 0 00-.714 1.7.75.75 0 01-1.072 1.05A2.495 2.495 0 012 11.5v-9zm10.5-1V9h-8c-.356 0-.694.074-1 .208V2.5a1 1 0 011-1h8zM5 12.25v3.25a.25.25 0 00.4.2l1.45-1.087a.25.25 0 01.3 0L8.6 15.7a.25.25 0 00.4-.2v-3.25a.25.25 0 00-.25-.25h-3.5a.25.25 0 00-.25.25z"
                fill-rule="evenodd"
              />
            </svg>
            <a
              href="https://github.com/marcosesperon/Messi"
            >
              marcosesperon/Messi
            </a>
          </p>
          <p>
            A simple message plugin for jQuery
          </p>
          <div>
            <p>
              <svg
                aria-label="star"
                data-view-component="true"
                height="16"
                role="img"
                version="1.1"
                viewBox="0 0 16 16"
                width="16"
              >
                <path
                  d="M8 .25a.75.75 0 01.673.418l1.882 3.815 4.21.612a.75.75 0 01.416 1.279l-3.046 2.97.719 4.192a.75.75 0 01-1.088.791L8 12.347l-3.766 1.98a.75.75 0 01-1.088-.79l.72-4.194L.818 6.374a.75.75 0 01.416-1.28l4.21-.611L7.327.668A.75.75 0 018 .25zm0 2.445L6.615 5.5a.75.75 0 01-.564.41l-3.097.45 2.24 2.184a.75.75 0 01.216.664l-.528 3.084 2.769-1.456a.75.75 0 01.698 0l2.77 1.456-.53-3.084a.75.75 0 01.216-.664l2.24-2.183-3.096-.45a.75.75 0 01-.564-.41L8 2.694v.001z"
                  fill-rule="evenodd"
                />
              </svg>
              143
            </p>
            <p>
              <span
                style="background-color: rgb(227, 5, 1);"
              />
              CSS
            </p>
            <p>
              Updated on 
              Wed Apr 27 2022
            </p>
          </div>
        </li>
        <li>
          <p>
            <svg
              aria-hidden="true"
              data-view-component="true"
              height="16"
              version="1.1"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M2 2.5A2.5 2.5 0 014.5 0h8.75a.75.75 0 01.75.75v12.5a.75.75 0 01-.75.75h-2.5a.75.75 0 110-1.5h1.75v-2h-8a1 1 0 00-.714 1.7.75.75 0 01-1.072 1.05A2.495 2.495 0 012 11.5v-9zm10.5-1V9h-8c-.356 0-.694.074-1 .208V2.5a1 1 0 011-1h8zM5 12.25v3.25a.25.25 0 00.4.2l1.45-1.087a.25.25 0 01.3 0L8.6 15.7a.25.25 0 00.4-.2v-3.25a.25.25 0 00-.25-.25h-3.5a.25.25 0 00-.25.25z"
                fill-rule="evenodd"
              />
            </svg>
            <a
              href="https://github.com/adityadroid/Messio"
            >
              adityadroid/Messio
            </a>
          </p>
          <p>
            An open source messenger app built using flutter
          </p>
          <div>
            <p>
              <svg
                aria-label="star"
                data-view-component="true"
                height="16"
                role="img"
                version="1.1"
                viewBox="0 0 16 16"
                width="16"
              >
                <path
                  d="M8 .25a.75.75 0 01.673.418l1.882 3.815 4.21.612a.75.75 0 01.416 1.279l-3.046 2.97.719 4.192a.75.75 0 01-1.088.791L8 12.347l-3.766 1.98a.75.75 0 01-1.088-.79l.72-4.194L.818 6.374a.75.75 0 01.416-1.28l4.21-.611L7.327.668A.75.75 0 018 .25zm0 2.445L6.615 5.5a.75.75 0 01-.564.41l-3.097.45 2.24 2.184a.75.75 0 01.216.664l-.528 3.084 2.769-1.456a.75.75 0 01.698 0l2.77 1.456-.53-3.084a.75.75 0 01.216-.664l2.24-2.183-3.096-.45a.75.75 0 01-.564-.41L8 2.694v.001z"
                  fill-rule="evenodd"
                />
              </svg>
              607
            </p>
            <p>
              <span
                style="background-color: rgb(159, 99, 32);"
              />
              Dart
            </p>
            <p>
              Updated on 
              Sun Nov 27 2022
            </p>
          </div>
        </li>
        <li>
          <p>
            <svg
              aria-hidden="true"
              data-view-component="true"
              height="16"
              version="1.1"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M2 2.5A2.5 2.5 0 014.5 0h8.75a.75.75 0 01.75.75v12.5a.75.75 0 01-.75.75h-2.5a.75.75 0 110-1.5h1.75v-2h-8a1 1 0 00-.714 1.7.75.75 0 01-1.072 1.05A2.495 2.495 0 012 11.5v-9zm10.5-1V9h-8c-.356 0-.694.074-1 .208V2.5a1 1 0 011-1h8zM5 12.25v3.25a.25.25 0 00.4.2l1.45-1.087a.25.25 0 01.3 0L8.6 15.7a.25.25 0 00.4-.2v-3.25a.25.25 0 00-.25-.25h-3.5a.25.25 0 00-.25.25z"
                fill-rule="evenodd"
              />
            </svg>
            <a
              href="https://github.com/Soumyajit2825/MESSI_MANIA"
            >
              Soumyajit2825/MESSI_MANIA
            </a>
          </p>
          <p>
            A simple fan made web-page of LIONEL MESSI
          </p>
          <div>
            <p>
              <svg
                aria-label="star"
                data-view-component="true"
                height="16"
                role="img"
                version="1.1"
                viewBox="0 0 16 16"
                width="16"
              >
                <path
                  d="M8 .25a.75.75 0 01.673.418l1.882 3.815 4.21.612a.75.75 0 01.416 1.279l-3.046 2.97.719 4.192a.75.75 0 01-1.088.791L8 12.347l-3.766 1.98a.75.75 0 01-1.088-.79l.72-4.194L.818 6.374a.75.75 0 01.416-1.28l4.21-.611L7.327.668A.75.75 0 018 .25zm0 2.445L6.615 5.5a.75.75 0 01-.564.41l-3.097.45 2.24 2.184a.75.75 0 01.216.664l-.528 3.084 2.769-1.456a.75.75 0 01.698 0l2.77 1.456-.53-3.084a.75.75 0 01.216-.664l2.24-2.183-3.096-.45a.75.75 0 01-.564-.41L8 2.694v.001z"
                  fill-rule="evenodd"
                />
              </svg>
              8
            </p>
            <p>
              <span
                style="background-color: rgb(171, 255, 33);"
              />
              HTML
            </p>
            <p>
              Updated on 
              Tue Sep 27 2022
            </p>
          </div>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul>
      <li>
        <p>
          <svg
            aria-hidden="true"
            data-view-component="true"
            height="16"
            version="1.1"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M2 2.5A2.5 2.5 0 014.5 0h8.75a.75.75 0 01.75.75v12.5a.75.75 0 01-.75.75h-2.5a.75.75 0 110-1.5h1.75v-2h-8a1 1 0 00-.714 1.7.75.75 0 01-1.072 1.05A2.495 2.495 0 012 11.5v-9zm10.5-1V9h-8c-.356 0-.694.074-1 .208V2.5a1 1 0 011-1h8zM5 12.25v3.25a.25.25 0 00.4.2l1.45-1.087a.25.25 0 01.3 0L8.6 15.7a.25.25 0 00.4-.2v-3.25a.25.25 0 00-.25-.25h-3.5a.25.25 0 00-.25.25z"
              fill-rule="evenodd"
            />
          </svg>
          <a
            href="https://github.com/marcosesperon/Messi"
          >
            marcosesperon/Messi
          </a>
        </p>
        <p>
          A simple message plugin for jQuery
        </p>
        <div>
          <p>
            <svg
              aria-label="star"
              data-view-component="true"
              height="16"
              role="img"
              version="1.1"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M8 .25a.75.75 0 01.673.418l1.882 3.815 4.21.612a.75.75 0 01.416 1.279l-3.046 2.97.719 4.192a.75.75 0 01-1.088.791L8 12.347l-3.766 1.98a.75.75 0 01-1.088-.79l.72-4.194L.818 6.374a.75.75 0 01.416-1.28l4.21-.611L7.327.668A.75.75 0 018 .25zm0 2.445L6.615 5.5a.75.75 0 01-.564.41l-3.097.45 2.24 2.184a.75.75 0 01.216.664l-.528 3.084 2.769-1.456a.75.75 0 01.698 0l2.77 1.456-.53-3.084a.75.75 0 01.216-.664l2.24-2.183-3.096-.45a.75.75 0 01-.564-.41L8 2.694v.001z"
                fill-rule="evenodd"
              />
            </svg>
            143
          </p>
          <p>
            <span
              style="background-color: rgb(227, 5, 1);"
            />
            CSS
          </p>
          <p>
            Updated on 
            Wed Apr 27 2022
          </p>
        </div>
      </li>
      <li>
        <p>
          <svg
            aria-hidden="true"
            data-view-component="true"
            height="16"
            version="1.1"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M2 2.5A2.5 2.5 0 014.5 0h8.75a.75.75 0 01.75.75v12.5a.75.75 0 01-.75.75h-2.5a.75.75 0 110-1.5h1.75v-2h-8a1 1 0 00-.714 1.7.75.75 0 01-1.072 1.05A2.495 2.495 0 012 11.5v-9zm10.5-1V9h-8c-.356 0-.694.074-1 .208V2.5a1 1 0 011-1h8zM5 12.25v3.25a.25.25 0 00.4.2l1.45-1.087a.25.25 0 01.3 0L8.6 15.7a.25.25 0 00.4-.2v-3.25a.25.25 0 00-.25-.25h-3.5a.25.25 0 00-.25.25z"
              fill-rule="evenodd"
            />
          </svg>
          <a
            href="https://github.com/adityadroid/Messio"
          >
            adityadroid/Messio
          </a>
        </p>
        <p>
          An open source messenger app built using flutter
        </p>
        <div>
          <p>
            <svg
              aria-label="star"
              data-view-component="true"
              height="16"
              role="img"
              version="1.1"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M8 .25a.75.75 0 01.673.418l1.882 3.815 4.21.612a.75.75 0 01.416 1.279l-3.046 2.97.719 4.192a.75.75 0 01-1.088.791L8 12.347l-3.766 1.98a.75.75 0 01-1.088-.79l.72-4.194L.818 6.374a.75.75 0 01.416-1.28l4.21-.611L7.327.668A.75.75 0 018 .25zm0 2.445L6.615 5.5a.75.75 0 01-.564.41l-3.097.45 2.24 2.184a.75.75 0 01.216.664l-.528 3.084 2.769-1.456a.75.75 0 01.698 0l2.77 1.456-.53-3.084a.75.75 0 01.216-.664l2.24-2.183-3.096-.45a.75.75 0 01-.564-.41L8 2.694v.001z"
                fill-rule="evenodd"
              />
            </svg>
            607
          </p>
          <p>
            <span
              style="background-color: rgb(159, 99, 32);"
            />
            Dart
          </p>
          <p>
            Updated on 
            Sun Nov 27 2022
          </p>
        </div>
      </li>
      <li>
        <p>
          <svg
            aria-hidden="true"
            data-view-component="true"
            height="16"
            version="1.1"
            viewBox="0 0 16 16"
            width="16"
          >
            <path
              d="M2 2.5A2.5 2.5 0 014.5 0h8.75a.75.75 0 01.75.75v12.5a.75.75 0 01-.75.75h-2.5a.75.75 0 110-1.5h1.75v-2h-8a1 1 0 00-.714 1.7.75.75 0 01-1.072 1.05A2.495 2.495 0 012 11.5v-9zm10.5-1V9h-8c-.356 0-.694.074-1 .208V2.5a1 1 0 011-1h8zM5 12.25v3.25a.25.25 0 00.4.2l1.45-1.087a.25.25 0 01.3 0L8.6 15.7a.25.25 0 00.4-.2v-3.25a.25.25 0 00-.25-.25h-3.5a.25.25 0 00-.25.25z"
              fill-rule="evenodd"
            />
          </svg>
          <a
            href="https://github.com/Soumyajit2825/MESSI_MANIA"
          >
            Soumyajit2825/MESSI_MANIA
          </a>
        </p>
        <p>
          A simple fan made web-page of LIONEL MESSI
        </p>
        <div>
          <p>
            <svg
              aria-label="star"
              data-view-component="true"
              height="16"
              role="img"
              version="1.1"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M8 .25a.75.75 0 01.673.418l1.882 3.815 4.21.612a.75.75 0 01.416 1.279l-3.046 2.97.719 4.192a.75.75 0 01-1.088.791L8 12.347l-3.766 1.98a.75.75 0 01-1.088-.79l.72-4.194L.818 6.374a.75.75 0 01.416-1.28l4.21-.611L7.327.668A.75.75 0 018 .25zm0 2.445L6.615 5.5a.75.75 0 01-.564.41l-3.097.45 2.24 2.184a.75.75 0 01.216.664l-.528 3.084 2.769-1.456a.75.75 0 01.698 0l2.77 1.456-.53-3.084a.75.75 0 01.216-.664l2.24-2.183-3.096-.45a.75.75 0 01-.564-.41L8 2.694v.001z"
                fill-rule="evenodd"
              />
            </svg>
            8
          </p>
          <p>
            <span
              style="background-color: rgb(171, 255, 33);"
            />
            HTML
          </p>
          <p>
            Updated on 
            Tue Sep 27 2022
          </p>
        </div>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`renders the repo-list component: no data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul />
    </div>
  </body>,
  "container": <div>
    <ul />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
